msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/stress/AbstractMsalUiStressTest.java

```diff
@@ -0,0 +1,267 @@
+
+
+        for (int i = 0; i < getNumberOfThreads(); i++) {
+            executorService.submit(new Runnable() {
+                @Override
+                public void run() {
+                    while (System.currentTimeMillis() - startTime < timeLimit && executionException == null) {
+                        try {
+                            S result = execute(prerequisites);
+
+                            boolean passed = isTestPassed(result);
+
+                            updateTestPassRate(passed);
+
+                            TimeUnit.MILLISECONDS.sleep(50);
+                        } catch (Exception ex) {
+                            executionException = ex;
+                        }
+                    }
+                }
+            });
+        }
+
+        executorService.shutdown();
+
+        // Force shut down of the executorService
+        if (!executorService.isShutdown()) {
+            executorService.shutdownNow();
+        }
+    }
+
+    private synchronized void updateTestPassRate(boolean passed) {
+        if (passed) {
+            testsPassed.incrementAndGet();
+        } else {
+            testsFailed.incrementAndGet();
+        }
+    }
+
+    private Runnable getPerformanceStatsCollector() {
+        return new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    while (true) {
+                        // time   cpu_usage   memory_usage   data_received   data_sent   num_threads   time_limit   device_memory   device_name   tests_passed   tests_failed
+                        String line = String.format(
+                                "%s,%.2f,%d,%d,%d,%d,%d,%d,%s,%d,%d",
+                                new SimpleDateFormat(DATE_FORMAT).format(new Date()),
+                                DeviceMonitor.getCpuUsage(),
+                                DeviceMonitor.getMemoryUsage(),
+                                DeviceMonitor.getNetworkTrafficInfo().getDiffBytesReceived(),
+                                DeviceMonitor.getNetworkTrafficInfo().getDiffBytesSent(),
+                                getNumberOfThreads(),
+                                getTimeLimit(),
+                                DeviceMonitor.getTotalMemory(),
+                                DeviceMonitor.getDeviceName(),
+                                testsPassed.get(),
+                                testsFailed.get()
+                        );
+
+                        writeFile(line);
+
+                        TimeUnit.MILLISECONDS.sleep(DEVICE_MONITOR_INTERVAL);
+                    }
+                } catch (Exception ex) {
+                    executionException = ex;
+                }
+            }
+        };
+    }
+
+    private synchronized void writeFile(final String output) {
+        try {
+            final FileAppender fileAppender = new FileAppender(getOutputFileName(), new LogcatLikeFormatter());
```

Comment:
Do you want to declare one of these outside, and just use it until the thread terminated, at which point you close it either via finally or after a latch release?